Improved accessibility attributes #1650

verheyenkoen opened the pull request:
Altered the accessibility attributes logic as follows:

* If the provided help text does not contain HTML, we render the `aria-describedby` attribute with a reference to the help element.
* If the provided help text does contain HTML, we render the `aria-details` attribute with a reference to the help element, as well as an `aria-description` attribute with the same text without HTML (except for hyperlinks, where the `href` is added as a suffix to the text contained in the node).

FIXES #1650

mietcls reviewed (commented):
The only ones that don't work are the tags:
It does not read the label out loud, nor the aria-describedby. Perhaps we need to restructure it a bit to be accessible.
<img width="1168" alt="Screenshot 2024-07-12 at 10 01 38" src="https://github.com/user-attachments/assets/43801050-72b8-4242-8510-d42508505790">
<img width="1158" alt="Screenshot 2024-07-12 at 09 46 03" src="https://github.com/user-attachments/assets/494bdee4-3ab3-4639-9d56-549154221f59">

verheyenkoen commented:
@mietcls I added some script to copy any `aria-*` attributes from the hidden `<textarea>` we generate to the actual Tagify DOM element `<span contenteditable>` and also added an `aria-label` for this one as the link with the corresponding `<label>` is broken (by design) so this has the same behaviour of first announcing the label and then the description. 

However, some of the information in the help text is lost in voiceover:
> Type keywords and separate them by TAB, ; or ENTER. A keyword is accepted when it turns blue. Duplicates are removed. e.g. DNA;Health Insurance<ENTER>

E.g. in this case, the semicolon is not spoken but is relevant (also if we put it between quotes). I guess we'd have to make that "semicolon" to be spoken. Obviously the sentence "A keyword is accepted when it turns blue." doesn't make a lot of sense without vision, but it would probably take us too far to have separate details and also, it could still be somewhat visible to some users using voiceover tools.

mietcls commented:
Thanks, the copy has to be revised anyways (it can be shorter) so I can keep it in mind.
